<launch>    
   <arg name="do_visualization" value="true"/>
   <arg name="enable_imu_preint_initial_guess" value="true"/>
   <param name="enable_imu_preint_initial_guess" type="bool" value="$(arg enable_imu_preint_initial_guess)"/>
   <param name="lidar_topic_name" type="string" value="/velodyne_points2"/>
   <param name="icp_result_visualization" type="bool" value="$(arg do_visualization)" />
   <param name="lidar_height_to_gps" type="double" value="0.14"/>
   <param name="icp_downsample_size" type="double" value="0.5"/>
   <param name="map_path" type="string" value="/home/gi/Downloads/GAAS_contrib_resources/map/airsim_msbuilding/output.pcd"/>
   <!--icp params-->

   <param name="icp_transformation_epsilon" type="double" value="0.05"/>
   <param name="icp_resolution" type="double" value="10.0"/>
   <param name="icp_max_correspondence_distance" value="8.0"/>
   <param name="icp_euclidean_fitness_epsilon" value="0.5"/>

   <node name="icp_matching_node" pkg="icp_lidar_localization" type="icp_matching_node" output="screen"/>
   <node name="icp_map_publiser_node" pkg="icp_lidar_localization" type="icp_map_publisher" output="screen"/>
   <group if="$(arg do_visualization)">
       <node name="rviz" pkg="rviz" type="rviz" args="-d $(find icp_lidar_localization)/launch/icp_localization.rviz" required="false"/>
   </group>
</launch>
